lib/internal/test_runner/runner.js

```diff
@@ -418,7 +418,22 @@ function watchFiles(testFiles, opts) {
   const filesWatcher = { __proto__: null, watcher, runningProcesses, runningSubtests };
   opts.root.harness.watching = true;
 
-  watcher.on('changed', ({ owners }) => {
+  watcher.on('changed', ({ owners, eventType }) => {
+    if (eventType === 'rename') {
+      const updatedTestFiles = createTestFileList();
+
+      const { 0: newFileName } = updatedTestFiles.filter((x) => !testFiles.includes(x));
```

Comment:
If we only care about the first index why not use `find`?
```suggestion
      const newFileName = ArrayPrototypeFind(updatedTestFiles, (x) => !ArrayPrototypeIncludes(testFiles, x));
```

Reply:
Good call, thanks!

Can I use this opportunity to ask why do we use `ArrayPrototypeFind` from `primordials` instead of directly use the `.find` syntax. I read from `doc/contributing/primordials.md` and what my understanding is that functions from `primordials` are from vm, so there is no look up needed in the prototype chain and cannot be mutated by the user, not sure if my understanding is correct or not.

Reply:
They are not from `node:vm`, they are defined in `lib/internal/per_context/primordials.js`. Using `.find` only works when the `Array` prototype wasn't tampered with in userland – which is usually a fair assumption, but for the test runner we don't want to make such assumption.

Reply:
got it, thanks!